sceptre/config/reader.py

```diff
@@ -352,11 +374,8 @@ def _read(self, rel_path, base_config=None):
 
         # Parse and read in the config files.
         this_config = self._recursive_read(directory_path, filename, config)
-
-        if "dependencies" in config or "dependencies" in this_config:
-            this_config["dependencies"] = CONFIG_MERGE_STRATEGIES["dependencies"](
-                this_config.get("dependencies"), config.get("dependencies")
-            )
+        # Apply merge strategies with the config that includes base_config values.
+        this_config.update(self._get_merge_with_stratgies(config, this_config))
```

Comment:
"dependencies" is handled in the common merge function.

This change is safe because all default strategies except for dependencies are `child_wins`, so behaviour doesn't change without the user selecting a strategy.